internal/standby/standby.go

```diff
@@ -75,13 +85,72 @@ func (s *Service) RunMQTT(ctx context.Context) error {
 			}
 		})
 
+	s.subscribeToTopic(s.cfg.MQTT.ReadCommandTopic,
+		func(client mqtt.Client, msg mqtt.Message) {
+			s.logger.Debug(fmt.Sprintf("Received message: %s from topic: %s", msg.Payload(), msg.Topic()))
+			s.LatestCommandReceived = time.Now()
+		})
+
 	return nil
 }
 
 func (s *Service) StopMQTT() {
 	s.MQTTClient.Disconnect(uint(1000))
 }
 
+func (s *Service) RunDetector(ctx context.Context) {
+	checkInterval := time.Duration(s.cfg.Standby.CheckInterval) * time.Second
+	outageDuration := time.Duration(s.cfg.Standby.OutageInterval) * time.Second
+
+	// initialise this so we don't immediately go into failure mode on startup
+	s.LatestCommandReceived = time.Now()
+
+	ticker := time.NewTicker(checkInterval)
+	for {
+		select {
+		case <-ticker.C:
+			s.CheckForOutage(outageDuration)
+		case <-ctx.Done():
+			ticker.Stop()
+
+			s.logger.Info("Shutting down detector")
+
+			return
+		}
+	}
+}
+
+func (s *Service) CheckForOutage(outageDuration time.Duration) {
+	timeSinceLastCmd := time.Since(s.LatestCommandReceived)
+	s.logger.Debug("checking", "time since last command", timeSinceLastCmd)
+	if timeSinceLastCmd > outageDuration {
+		if s.Mode == StandbyMode {
+			s.logger.Info("Outage detected", "config duration", outageDuration, "time since last command", timeSinceLastCmd)
+			s.Mode = CommandMode
+		} else {
+			s.logger.Debug("Ongoing outage", "config duration", outageDuration, "time since last command", timeSinceLastCmd)
+		}
+		// TODO: Trigger a process to identify closest optimisation command from plan and send it
+	} else {
+		if s.Mode == CommandMode {
+			s.logger.Info("Commands resumed after outage", "time since last command", timeSinceLastCmd)
+			s.Mode = StandbyMode
+		}
+	}
+}
+
+func (s *Service) Start(ctx context.Context) error {
+	if err := s.RunMQTT(ctx); err != nil {
+		return fmt.Errorf("starting MQTT client: %w", err)
+	}
+	s.RunDetector(ctx)
```

Comment:
I think you meant `go s.RunDetector(ctx)`? Current invocation blocks on the infinite ticker loop.

Reply:
yes, thanks for pointing this out 🙌